### PR TITLE
Use the data directory also for keys in aea manager

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Use the data directory also for keys in the AEA manager.

## Fixes

Continuation of #2228 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
n/a

### CODE REVIEW

File: aea/aea_builder.py [hunk 1]
```diff
@@ -1126,9 +1155,7 @@ def build(self, connection_ids: Optional[Collection[PublicId]] = None,) -> AEA:
         datadir = self._get_data_dir()
         self._check_we_can_build()
         logging.config.dictConfig(self._logging_config)
-        wallet = Wallet(
-            copy(self.private_key_paths), copy(self.connection_private_key_paths)
```
Comment (DavidMinarsch) [inline aea/aea_builder.py:1130]: what about copy?

File: aea/manager/project.py [hunk 2]
```diff
@@ -194,7 +191,7 @@ def _create_private_key(
             if is_connection
             else f"{self.agent_name}_{ledger}_private.key"
         )
```
Comment (DavidMinarsch) [inline aea/manager/project.py:193]: these file names no longer need to be custom now

File: aea/aea_builder.py (hunk 1 above)
Comment (marcofavorito) [reply aea/aea_builder.py:1130]: not needed in dict comprehension

File: aea/manager/project.py (hunk 2 above)
Comment (marcofavorito) [reply aea/manager/project.py:193]: yes

File: aea/manager/manager.py [hunk 3]
```diff
@@ -323,7 +311,6 @@ def stop_manager(
     def _cleanup(self, only_keys: bool = False) -> None:
         """Remove workdir if was created."""
         if only_keys:
```
Comment (DavidMinarsch) [inline aea/manager/manager.py:313]: must be called `only_data` then

File: tests/test_manager/test_manager.py [hunk 4]
```diff
@@ -73,11 +73,11 @@ def test_workdir_created_removed(self, *args):
         assert not os.path.exists(self.working_dir)
         assert not os.path.exists(self.working_dir)
 
-    def test_keys_dir_presents(self, *args):
+    def test_data_dir_presents(self, *args):
         """Check not fails on exists key dir."""
         try:
             os.makedirs(self.working_dir)
-            os.makedirs(self.manager._keys_dir)
+            os.makedirs(self.manager._data_dir)
```
Comment (DavidMinarsch) [inline tests/test_manager/test_manager.py:80]: imo the manager should make it

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**